lib/galaxy/tools/imp_exp/__init__.py

```diff
@@ -395,6 +406,27 @@ def default(self, obj):
                     return rval
                 return json.JSONEncoder.default(self, obj)
 
+        class CollectionsEncoder(json.JSONEncoder):
+            """ Custom JSONEncoder for a Collection. """
+
+            def default(self, obj):
+                """ Encode a collection, default encoding for everything else. """
+                if isinstance(obj, trans.model.HistoryDatasetCollectionAssociation):
+                    # dump attrs of each dataset of this collection
+                    collections_datasets_attrs = []
+                    collection_datasets = obj.dataset_instances
```

Comment:
This really flattens out the dataset collection - so you loose all collection nesting informations and element identifiers that are very important I think. You want to somehow recursively encode ``object.collection`` - the ``DatasetCollection`` object. This will include the ``collection_type`` and element identifiers.

Reply:
ok, I've got collection_type now. Thank you, I've missed the fact that I can acces the collection object itself!
But I'm not sure I understand well this part:

> You want to somehow recursively encode object.collection - the DatasetCollection object. This will include the collection_type and element identifiers.

I've tried, I retrieve indeed `DatasetCollection` object. `DatasetCollection` gives me a lot of elements (state, populated) and objects (dataset_instances and dataset_elements) but what do you think I will need to export?
There is already a lot of data in my file:
[collections_attrs.txt](https://github.com/galaxyproject/galaxy/files/1269883/collections_attrs.txt)

The related dataset files are stored in a directory (collections_datasets, maybe it could fit in the already created datasets directory but it helps me right now to see if I have of the files I need)